azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageVersionDaoImpl.java

```diff
@@ -354,8 +354,27 @@ public void updateImageVersion(final ImageVersion imageVersionRequest)
       queryBuilder.append(" where iv.id = ? and iv.type_id = it.id and it.name = ?");
       params.add(imageVersionRequest.getId());
       params.add(imageVersionRequest.getName());
-      final int updateCount = this.databaseOperator.update(queryBuilder.toString(),
-          Iterables.toArray(params, Object.class));
+      int updateCount = 0;
+      if (imageVersionRequest.getState() != State.ACTIVE) {
+        updateCount = this.databaseOperator.update(queryBuilder.toString(),
+            Iterables.toArray(params, Object.class));
+      } else {
+        // If the state is being set to ACTIVE then only the current image version can be set
+        // to ACTIVE and the rest of the image versions with ACTIVE state must be set to STABLE.
+        final SQLTransaction<Integer> setCurrentVersionActive = transOperator -> {
+          // Set all the current active versions to STABLE
+          final String setStable = String.format("update image_versions iv, image_types it " +
+              " set iv.state = 'STABLE' where iv.type_id = it.id and it.name = '%s' "
```

Comment:
Please use enum name instead of string for STABLE state

Reply:
Will create a followup PR to make the change.